thunder/tests/test_auto_register_torchops.py

```diff
@@ -0,0 +1,197 @@
+from functools import partial
+from unittest.mock import patch
+
+import pytest
+import thunder
+import torch
+
+from thunder.tests.framework import requiresCUDA, TorchExecutor
+from thunder.tests.make_tensor import make_tensor
+from thunder.tests.opinfos import get_opinfo, OpInfo
+from thunder.tests.test_einops import skipIfNoCUDA
+
+
+@skipIfNoCUDA
+@pytest.mark.parametrize("requires_grad", [True, False], ids=("train", "inference"))
+@pytest.mark.parametrize("device,", ["cuda", "cpu"])
+def test_torch_ops_trace(device, requires_grad):
+    from itertools import islice
+
+    import thunder.torch.default_torch_ops as ops
+    from torch.testing._internal.common_methods_invocations import op_db
+
+    name2func = {}
+    [name2func.setdefault(v.__name__, v) for v in ops.torch_auto_registered_ops[torch]]
+    [name2func.setdefault(f"nn.functional.{v.__name__}", v) for v in ops.torch_auto_registered_ops[torch.nn.functional]]
+    # Use the sample input from torch.xx to test torch.tensor.xx
+    [name2func.setdefault(f"Tensor.{v.__name__}", v) for v in ops.torch_auto_registered_ops[torch.Tensor]]
+    print(f"auto reg ops: {len(name2func)}")
+    op_infos = [opinfo for opinfo in op_db if opinfo.name in name2func.keys()]
+    total = 0
+    cnt = 0
+    suc = 0
+    for op_info in op_infos:
+        if device == "cuda" and torch.float32 not in op_info.dtypesIfCUDA:
+            continue
+        if device == "cpu" and not torch.float32 in op_info.dtypes:
+            continue
+
+        # No cuda backend support
+        if op_info.name in ("nonzero_static",) and device == "cuda":
+            continue
+        # RuntimeError: Calling torch.linalg.cholesky on a CPU tensor requires compiling PyTorch with LAPACK.
+        if op_info.name in ("cholesky",) and device == "cpu":
+            continue
+        # (t6, t7, t8, t9, t10, t11) = torch.histogramdd(t_0, (1, 1, 1, 1, 1), weight=None, density=False)
+        # ValueError: not enough values to unpack (expected 6, got 2)
```

Comment:
could you tell me what these lines mean? I couldn't get my head around

Reply:
After checking I find this operator is not supported by the vjp, it reports TypeError when cpu-train and the above error when cpu-inference, I'll remove this from the op list